sphinx_needs/directives/need.py

```diff
@@ -361,6 +363,16 @@ def analyse_need_locations(app: Sphinx, doctree: nodes.document) -> None:
             need_info["parent_needs"] = parent_needs
             need_info["parent_need"] = parent_needs[0]
 
+        # find any parent only expressions, and note them on the need data item
+        expressions = []
+        parent: nodes.Element = need_node
+        while parent := getattr(parent, "parent", None):  # type: ignore
+            if isinstance(parent, addnodes.only):  # noqa: SIM102
+                if expr := parent.get("expr"):
+                    expressions.append(expr)
```

Comment:
Shouldn't we add a `break` here, so that we don't "look up" anymore after `only` was found?

Reply:
no; we want to capture ALL parent `only` expressions, for example (this is in the test case):

```restructuredtext
.. only:: not something

   .. only:: other

      .. req:: Requirement 4
         :id: REQ_4
```

here the requirement should only be kept if BOTH `not something` and `other` evaluate to true,
so we capture and store `{"only_expressions": ["not something", "other"]}`

Reply:
good point and fully agree :+1: